src/execution/operator/order/physical_top_n.cpp

```diff
@@ -443,31 +443,31 @@ void TopNHeap::Reduce() {
 
 void TopNHeap::InitializeScan(TopNScanState &state, bool exclude_offset) {
 	auto heap_copy = heap;
-	// traverse the rest of the heap
 	state.scan_order.resize(heap_copy.size());
-	while (!heap_copy.empty()) {
-		std::pop_heap(heap_copy.begin(), heap_copy.end());
-		state.scan_order[heap_copy.size() - 1] = UnsafeNumericCast<sel_t>(heap_copy.back().index);
-		heap_copy.pop_back();
+
```

Comment:
Just for my understanding, should there not be a sort here? I am not sure I see where the sort happens.

Reply:
The sort happens in `Finalize()`, I can clarify with a comment

Reply:
All good, thanks